### PR TITLE
tests: add mint log support for minio-java functional tests

### PR DESCRIPTION
**Pull request by kannappanr**
In Progress: Update minio-java to output new mint log format

https://github.com/minio/minio-java/issues/597

### CODE REVIEW

File: functional/FunctionalTest.java [hunk 1]
```diff
@@ -137,72 +140,264 @@ public static String getRandomName() {
...
+      if (mintEnv) {
+        System.out.println( mintLog("makeBucket(String bucketName)", 
+                                      null, 
+                                      System.currentTimeMillis() - startTime,
+                                      pass,
+                                      null,
+                                      null,
+                                      null));
+
+      }
```
Comment (balamurugana) [inline functional/FunctionalTest.java]: You can move this `if` statement into `mintLog()`

File: functional/MintLogger.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  /**
+    * Constructor.
+    **/
+  public MintLogger(String function,
+                    String args,
+                    long duration,
+                    String status,
+                    String alert,
+                    String message,
+                    Exception e) {
```
Comment (balamurugana) [inline functional/MintLogger.java]: If possible, you could return error message as string than `Exception`

File: functional/FunctionalTest.java [hunk 3]
```diff
@@ -44,6 +44,9 @@
 
 
 public class FunctionalTest {
+  private static final String pass = "Pass";
+  private static final String failed = "Fail";
+  private static final String na = "NA";
```
Comment (balamurugana) [inline functional/FunctionalTest.java]: You could capitalize the value and name here

**Comment by NitishT** [comment]
@kannappanr is #602 different from this?

**Comment by balamurugana** [comment]
@kannappanr can you fix commit log and title of this PR to indicate functional test logs for mint like
`tests: add mint log support for functional tests`

File: functional/FunctionalTest.java [hunk 4]
```diff
@@ -137,72 +140,244 @@ public static String getRandomName() {
...
+      client.makeBucket(name, "eu-central-1");
+      client.removeBucket(name);
+
+      mintLog("makeBucket(String bucketName, String region)", 
+                          "name: " + name + ", region: eu-central-1", 
+                          System.currentTimeMillis() - startTime,
+                          PASS,
+                          null,
+                          null,
+                          null);
```
Comment (NitishT) [inline functional/FunctionalTest.java]: Passing args in new line seems a little weird to me. Any specific reasons for doing this? If not we can follow the style we follow for other API calls - `client.makeBucket(name, "eu-central-1");`

Comment (balamurugana) [inline functional/FunctionalTest.java]: @kannappanr I think you could add methods like mintLogPass() and mintLogFail() and use them respectively

**Pull request merged**